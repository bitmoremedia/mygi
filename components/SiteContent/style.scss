@import '../../css/variables';

.site-content {
  position: relative;
  display: block;
  overflow: hidden;
  width: 100%;
  padding-left: 10px;
  padding-right: 10px;
  @include respond-to(tablet) {
    padding-left: 40px;
  }
  @include respond-to(laptop) {
    padding-left: 60px;
  }
  @include respond-to(desktop) {
    padding-left: 80px;
  }
}
.site-content__page {
  position: relative;
  width: 100%;
  max-width: 1000px;
  margin: 0 auto;
}
