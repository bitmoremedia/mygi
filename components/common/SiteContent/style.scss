@import '../../../css/variables';

.site-content {
  position: relative;
  display: block;
  overflow: hidden;
  width: 100%;
  padding-top: 10px;
  padding-left: 15px;
  padding-bottom: 10px;
  padding-right: 15px;
  @include respond-to(tablet) {
    padding-left: 40px;
    padding-right: 40px;
  }
  @include respond-to(laptop) {
    padding-left: 60px;
    padding-right: 60px;
  }
  @include respond-to(desktop) {
    padding-left: 80px;
    padding-right: 80px;
  }
}
.site-content__inner {
  position: relative;
  width: 100%;
  max-width: 1000px;
  margin: 0 auto;
}
