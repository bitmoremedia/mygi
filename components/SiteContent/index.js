import React from 'react';

import './style';

const SiteContent = ({children}) => {

  return (
    <div className='site-content'>
      <div className='site-content__page'>
        {children}
      </div>
    </div>
  );
};

export default SiteContent;
