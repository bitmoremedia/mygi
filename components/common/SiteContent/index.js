import React from 'react';
import classNames from 'classnames';

import './style';

const SiteContent = ({children, page, className}) => {

  const wrapperClass = classNames('site-content', { [className]: className });

  if (page){
    return (
      <div className={wrapperClass}>
        <div className='site-content__page'>
          <div className='site-content__inner'>
            {children}
          </div>
        </div>
      </div>
    );
  }

  return (
    <div className={wrapperClass}>
      <div className='site-content__inner'>
        {children}
      </div>
    </div>
  );
};

export default SiteContent;
