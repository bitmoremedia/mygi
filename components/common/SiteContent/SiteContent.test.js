import React from 'react'
import renderer from 'react-test-renderer'

import SiteContent from './index'

test('basic snapshot render', () => {
  const component = renderer.create(
    <SiteContent>
      <div className="some-class">some content</div>
    </SiteContent>,
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('wraps the child content as a page if specified', () => {
  const component = renderer.create(
    <SiteContent page>
      <div className="some-class">some content</div>
    </SiteContent>,
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('wraps content in wrapper class if provided', () => {
  const component = renderer.create(
    <SiteContent className="some-wrapper-class">
      <div className="some-class">some content</div>
    </SiteContent>,
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
