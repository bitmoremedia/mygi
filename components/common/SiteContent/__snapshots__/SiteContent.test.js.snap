// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic snapshot render 1`] = `
<div
  className="site-content"
>
  <div
    className="site-content__inner"
  >
    <div
      className="some-class"
    >
      some content
    </div>
  </div>
</div>
`;

exports[`wraps content in wrapper class if provided 1`] = `
<div
  className="site-content some-wrapper-class"
>
  <div
    className="site-content__inner"
  >
    <div
      className="some-class"
    >
      some content
    </div>
  </div>
</div>
`;

exports[`wraps the child content as a page if specified 1`] = `
<div
  className="site-content"
>
  <div
    className="site-content__page"
  >
    <div
      className="site-content__inner"
    >
      <div
        className="some-class"
      >
        some content
      </div>
    </div>
  </div>
</div>
`;
